File: src/primus/pawel/Exercise_198.java
package primus.pawel;
// Halving SUM
// 7kyu
// https://www.codewars.com/kata/5a58d46cfd56cb4e8600009d/train/java
// 2021-08-25 T:01:35:18
class Exercise_198 {

    public  static int halvingSum(int n) {
        int number = n;
        int sum = 0;

        while (number!=1){
            sum += number;
            number = number/2;
        }
        return sum + 1;
    }


}

/*      int halvingSum(int n) {
    return IntStream.iterate(n, x -> x != 0, x -> x / 2).sum();
  }
  */

File: src/JS/Exercise_198.js
// Find Duplicates
// 7kyu
// https://www.codewars.com/kata/5558cc216a7a231ac9000022/train/javascript
// 2022-08-13 T:14:18:20

function duplicates(arr) {
    var result = [];

    for (let i = 0; i < arr.length; i++) {

        for (let j = i - 1; j >= 0; j--) {

            if (arr[i] === arr[j]) {
                result.push(arr[i]);
                break;
            }
        }
    }

    return [...new Set(result)];
}

console.log(duplicates([1, 2, 4, 4, 3, 3, 1, 5, 3 ]))
